### PR TITLE
(CFACT-169) Provide ruby during cfacter build time where applicable

### PR DESCRIPTION
**Pull request by melissaanne**
With this PR, we're bringing in ruby and ruby-devel to have available during build time. This will let us take advantage of the work done in https://github.com/puppetlabs/cfacter/pull/255 so we can provide the cfacter ruby library, as well as run the full test suite, which will only include the tests against ruby when when a sufficient version of ruby is available.

### CODE REVIEW

**Comment by melissaanne** [comment]
This PR depends on work in https://github.com/puppetlabs/cfacter/pull/255

File: ext/debian/rules [hunk 1]
```diff
@@ -32,5 +32,8 @@ override_dh_auto_install:
 	$(MAKE) -j$(NUMPROC) install DESTDIR=$(CURDIR)/debian/tmp
 	rm -r $(CURDIR)/debian/tmp$(PREFIX)/include/boost
 
+override_dh_auto_test:
+	$(BUILDPREFIX)/bin/ctest -V
```
Comment (MikaelSmith) [inline ext/debian/rules]: Does this depend on #255?

Comment (melissaanne) [inline ext/debian/rules]: Probably not, actually.

File: ext/redhat/cfacter.spec.erb [hunk 2]
```diff
@@ -22,6 +27,11 @@ Source0:        http://puppetlabs.com/downloads/%{name}/%{name}-%{realversion}.t
 
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
+# Only >=el7 and fedora have a succifient version of ruby available
```
Comment (shrug) [inline ext/redhat/cfacter.spec.erb]: Nit: misspelled 'sufficient'

Comment (melissaanne) [inline ext/redhat/cfacter.spec.erb]: I'm really bad at spelling...

File: ext/redhat/cfacter.spec.erb [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
```
Comment (shrug) [inline ext/redhat/cfacter.spec.erb]: Nit: misspelled 'sufficient'

Comment (melissaanne) [inline ext/redhat/cfacter.spec.erb]: I FIXED IT!

**Comment by shrug** [comment]
:+1:  pending tests

**Pull request merged**